ledger-api/grpc-definitions/com/daml/ledger/api/v1/completion.proto

```diff
@@ -32,6 +36,51 @@ message Completion {
   // Optional
   string transaction_id = 3;
 
+  // The application ID that was used for the submission, as described in ``commands.proto``.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Optional for historic completions where this data is not available.
+  string application_id = 6;
+
+  // The set of parties on whose behalf the commands were executed.
+  // Contains the union of ``party`` and ``act_as`` from ``commands.proto``.
+  // The order of the parties need not be the same as in the submission.
+  // Each element must be a valid PartyIdString (as described in ``value.proto``).
+  // Optional for historic completions where this data is not available.
+  repeated string act_as = 7;
+
+  // The submission ID this completion refers to, as described in ``commands.proto``.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Optional for historic completions where this data is not available.
+  string submission_id = 4;
+
+  // The submission rank of the submission this completion refers to, as described in ``commands.proto``.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Optional for historic completions where this data is not available.
+  string submission_rank = 5;
+
```

Comment:
```suggestion
   // Record time of this completion.
   google.protobuf.Timestamp record_time = 6;
```

Reply:
Isn't the record time communicated as part of the Checkpoint that contains the completions?

Reply:
I've never understood why we distinguish between completion events and checkpoints. My current understanding is that we emit a checkpoint for every completion, but I may well be wrong.